Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -883,3 +894,20 @@ internal func assertExpression(_ body: () -> Bool) {
         return body()
     }())
 }
+
+extension SelectableEventLoop: NIOCustomTimerImplementation {
+    @inlinable
+    internal func setTimer(for deadline: NIODeadline, _ handler: any NIOTimerHandler) -> UInt64 {
+        let taskId = self.scheduledTaskCounter.loadThenWrappingIncrement(ordering: .relaxed)
```

Comment:
We now have two places here that create a task id and then a `ScheduledTask` can we de-dupe the code into a shared method?

Reply:
I wanted to keep the `CustomTimerImplementation` conformance in its own extension here for clarity but, yes, let me take a look and see what refactoring I can do in the `SelectableEventLoop` to provide a reusable method.